Start removing implicit any errors

smfoote opened the pull request:
Thusfar:

component-managers/abstract.ts (2 errors remain)
component-managers/curly.ts (some errors remain)
component-managers/mount.ts
utils/curly-component-state-bucket.ts

Commit:
Start removing implicit any errors

Thusfar:

component-managers/abstract.ts (2 errors remain)
component-managers/curly.ts (some errors remain)
component-managers/mount.ts
utils/curly-component-state-bucket.ts

krisselden commented on packages/container/lib/index.d.ts:
```diff
@@ -0,0 +1,17 @@
+interface Container {
```
I stubbed out a more limited interface for this in ember-glimmer we should unify

krisselden commented on packages/ember-glimmer/lib/component-managers/curly.ts:
```diff
@@ -74,26 +91,26 @@ function applyAttributeBindings(element, attributeBindings, component, operation
   public template: any;
 
-  constructor(template) {
+  constructor(template: any) {
```
I did make a type for template

krisselden commented on packages/ember-glimmer/lib/component-managers/curly.ts:
```diff
@@ -74,26 +91,26 @@ function applyAttributeBindings(element, attributeBindings, component, operation
 }
 
-function ariaRole(vm) {
+function ariaRole(vm: any) {
```
there is a type exported for this, I think it is called PublicVM

smfoote replied:
Works for the `ariaRole` function, but not `tagName`: `Property 'view' does not exist on type 'DynamicScope'`

krisselden commented on packages/ember-glimmer/lib/component-managers/curly.ts:
```diff
@@ -107,22 +124,22 @@ export class PositionalArgumentReference {
   }
 
-  get(key) {
+  get(key: any) {
```
string

krisselden commented on packages/ember-glimmer/lib/component-managers/curly.ts:
```diff
@@ -231,16 +248,15 @@ export default class CurlyComponentManager extends AbstractManager<ComponentStat
 
-  templateFor(component, env) {
+  templateFor(component: ComponentStateBucket, env: Environment): any {
     let Template = get(component, 'layout');
```
we have a type for what templates return look at the template .d.ts I updated

krisselden commented on packages/ember-routing/lib/index.d.ts:
```diff
@@ -0,0 +1,16 @@
+export const HashLocation: any;
+export const HistoryLocation: any;
+export const AutoLocation: any;
+
```
lets at least type the signature of the functions

Commit:
Add better types for ember-routing

Commit:
Use WrappedTemplateFactory as template type

Commit:
Use actual types (mostly) for component-managers/

There are a few uses of `any` still, and a few errors I'm still trying
to figure out.

Commit:
Types for /components and some of /helpers

Commit:
Add types to the rest of /helpers

Commit:
Better types for /modifiers /syntax /utils

Commit:
Update types for remaining files.

Tests are broken due at the moment. CurlyComponentManager's templateFor
is where the problems begin. Fix will come in next commit.

Commit:
Merge branch 'master' into implicit-any

Commit:
Revert signatures to templateFor and layoutFor

Commit:
Fix arguments test failures

smfoote commented:
Despite the green check, I don't think this is ready yet. There are several TypeScript errors still.

rwjblue commented:
@smfoote - Sounds good, let us know when you think it is ready for another round of review and possible merging.

smfoote commented:
Since the checks are passing, we might merge this one as is then fix the remaining TypeScript errors in a future PR. I'll leave that up to @krisselden
